libhsm/src/lib/libhsm.c

```diff
@@ -1382,7 +1382,7 @@ hsm_hex_unparse(char *dst, const unsigned char *src, size_t len)
     size_t i;
 
     for (i = 0; i < len; i++) {
-        snprintf(dst + (2*i), dst_len, "%02x", src[i]);
+        snprintf(dst + (2*i), dst_len - (2*i), "%02x", src[i]);
```

Comment:
Would it be clearer with this?
```suggestion
        snprintf(dst, dst_len, "%02x", src[i]);
        dst += 2;
        dst_len -= 2;
```